source/nodejs/tests/unit-tests/src/unit-tests.test.ts

```diff
@@ -14,3 +17,99 @@ describe("test Text property on TextBlock", () => {
         expect('{"type":"TextBlock","text":"Some text"}').toBe(JSON.stringify(textBlockJson));
     });
 });
+
+describe("Test Templating Library", () => {
+    it("BasicTemplate", () => {
```

Comment:
love these tests - I think I'd like to see a few additional scenarios covered:

* what happens if I pass `[ "foo", {"bar": "baz"}]` as my `$host`?
* how about `"just some string"`

Reply:
To clarify, $host and $root should only accept valid JSON, correct?

Also, did we want to accept host data without root data - don't remember what we decided on the call :)

Reply:
@paulcam206 I added the following tests

- BasicTemplate, only provide host data
- Host data given as string
- Host data given as array

I also went ahead and made both $host and $root optional - I can change this back if we want